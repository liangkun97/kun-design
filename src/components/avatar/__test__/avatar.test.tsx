import React from "react";
import { render, cleanup } from "@testing-library/react";
import Avatar, { AvatarSize } from "../index";

describe("test Avatar component", () => {
  it("should render default avatar", () => {
    const wrapper = render(<Avatar username="kun" data-testid="avatar-div" />);
    expect(wrapper).toMatchSnapshot();
    const div = wrapper.getByTestId("avatar-div");
    expect(div).toBeInTheDocument();
    const username = wrapper.getByText("k");
    expect(username).toBeTruthy();
  });
  it("should render correct size", () => {
    let wrapper = render(<Avatar data-testid="avatar-div" />);
    let div = wrapper.getByTestId("avatar-div");
    expect(div).toHaveStyle(`height:${AvatarSize.medium}px`);
    expect(div).toHaveStyle(`width:${AvatarSize.medium}px`);
    expect(div).toHaveStyle(`line-height:${AvatarSize.medium}px`);
    let username = wrapper.getByTestId("avatar-div");
    expect(username).toHaveStyle(`line-height:${AvatarSize.medium}px`);
    cleanup();
    wrapper = render(<Avatar size="large" data-testid="avatar-div" />);
    expect(wrapper).toMatchSnapshot();
    div = wrapper.getByTestId("avatar-div");
    expect(div).toHaveStyle(`height:${AvatarSize.large}px`);
    expect(div).toHaveStyle(`width:${AvatarSize.large}px`);
    expect(div).toHaveStyle(`line-height:${AvatarSize.large}px`);
    username = wrapper.getByTestId("avatar-div");
    expect(username).toHaveStyle(`line-height:${AvatarSize.large}px`);
    cleanup();
    wrapper = render(<Avatar size="small" data-testid="avatar-div" />);
    div = wrapper.getByTestId("avatar-div");
    expect(div).toHaveStyle(`height:${AvatarSize.small}px`);
    expect(div).toHaveStyle(`width:${AvatarSize.small}px`);
    expect(div).toHaveStyle(`line-height:${AvatarSize.small}px`);
    username = wrapper.getByTestId("avatar-div");
    expect(username).toHaveStyle(`line-height:${AvatarSize.small}px`);
    cleanup();
    wrapper = render(<Avatar size="tiny" data-testid="avatar-div" />);
    div = wrapper.getByTestId("avatar-div");
    expect(div).toHaveStyle(`height:${AvatarSize.tiny}px`);
    expect(div).toHaveStyle(`width:${AvatarSize.tiny}px`);
    expect(div).toHaveStyle(`line-height:${AvatarSize.tiny}px`);
    username = wrapper.getByTestId("avatar-div");
    expect(username).toHaveStyle(`line-height:${AvatarSize.tiny}px`);
    cleanup();
    wrapper = render(<Avatar size="medium" data-testid="avatar-div" />);
    div = wrapper.getByTestId("avatar-div");
    expect(div).toHaveStyle(`height:${AvatarSize.medium}px`);
    expect(div).toHaveStyle(`width:${AvatarSize.medium}px`);
    expect(div).toHaveStyle(`line-height:${AvatarSize.medium}px`);
    username = wrapper.getByTestId("avatar-div");
    expect(username).toHaveStyle(`line-height:${AvatarSize.medium}px`);
  });
  it("should correct loading", () => {
    let wrapper = render(<Avatar isLoading data-testid="avatar-div" />);
    expect(wrapper).toMatchSnapshot();
    let svg = wrapper.getByTestId("avatar-div");
    expect(svg).toBeVisible();
    cleanup();
    wrapper = render(
      <Avatar
        isLoading
        username="123"
        src="/"
        size="tiny"
        data-testid="avatar-div"
      />
    );
    svg = wrapper.getByTestId("avatar-div");
    expect(svg).toBeVisible();
  });
  it("should correct img", () => {
    let wrapper = render(
      <Avatar src="www.test.com" data-testid="avatar-div" />
    );
    let img = wrapper.getByTestId("avatar-img");
    expect(img.tagName).toEqual("IMG");
    expect(img).toHaveStyle("width:100%");
    expect(img).toHaveAttribute("src", "www.test.com");
    expect(img).toHaveAttribute("alt", "loading");
    cleanup();
    wrapper = render(
      <Avatar
        src="www.liang-kun.cn"
        username="liang-kun"
        data-testid="avatar-div"
      />
    );
    img = wrapper.getByTestId("avatar-img");
    expect(img).toHaveAttribute("src", "www.liang-kun.cn");
    expect(img).toHaveAttribute("alt", "liang-kun");
  });
  it("should render correct username", () => {
    let wrapper = render(
      <Avatar username="liang-kun" data-testid="avatar-div" />
    );
    expect(wrapper).toMatchSnapshot();
    const div = wrapper.getByTestId("avatar-div");
    expect(div).toHaveStyle("text-transform:uppercase");
    let username = wrapper.getByText("l");
    expect(username).toBeVisible();
    cleanup();
    wrapper = render(<Avatar username="中文汉字" data-testid="avatar-div" />);
    username = wrapper.getByText("中");
    expect(username).toBeTruthy();
  });
});
