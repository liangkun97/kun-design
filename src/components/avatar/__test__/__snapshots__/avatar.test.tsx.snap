// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Avatar component it should render correct size 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dkPtRN cOQEWS"
        data-testid="avatar-div"
      >
        <div
          aria-hidden="true"
          class="sc-hKwDye hxekdW"
        >
          l
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dkPtRN cOQEWS"
      data-testid="avatar-div"
    >
      <div
        aria-hidden="true"
        class="sc-hKwDye hxekdW"
      >
        l
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Avatar component it should render default avatar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dkPtRN gBCMJX"
        data-testid="avatar-div"
      >
        <div
          aria-hidden="true"
          class="sc-hKwDye kbA-dvU"
        >
          k
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dkPtRN gBCMJX"
      data-testid="avatar-div"
    >
      <div
        aria-hidden="true"
        class="sc-hKwDye kbA-dvU"
      >
        k
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Avatar component should correct loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dkPtRN ftuBVG"
        data-testid="avatar-div"
      >
        <svg
          class="sc-bdvvtL kipgBN"
          color="black"
          height="20"
          viewBox="0 0 1024 1024"
          width="20"
        >
          <path
            class="sc-gsDKAQ bvJhse"
            color="black"
            d="M533 960a850 850 0 0 0 386-92v-19c0-117-242-223-306-234-20-3-21-58-21-58s59-58 72-137c35 0 56-84 21-113 2-31 45-243-173-243S337 276 338 307c-34 29-13 113 22 113 13 79 72 137 72 137s-1 55-21 58c-64 11-301 115-306 231a855 855 0 0 0 428 114z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dkPtRN ftuBVG"
      data-testid="avatar-div"
    >
      <svg
        class="sc-bdvvtL kipgBN"
        color="black"
        height="20"
        viewBox="0 0 1024 1024"
        width="20"
      >
        <path
          class="sc-gsDKAQ bvJhse"
          color="black"
          d="M533 960a850 850 0 0 0 386-92v-19c0-117-242-223-306-234-20-3-21-58-21-58s59-58 72-137c35 0 56-84 21-113 2-31 45-243-173-243S337 276 338 307c-34 29-13 113 22 113 13 79 72 137 72 137s-1 55-21 58c-64 11-301 115-306 231a855 855 0 0 0 428 114z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Avatar component should render correct username 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dkPtRN gBCMJX"
        data-testid="avatar-div"
      >
        <div
          aria-hidden="true"
          class="sc-hKwDye kbA-dvU"
        >
          l
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dkPtRN gBCMJX"
      data-testid="avatar-div"
    >
      <div
        aria-hidden="true"
        class="sc-hKwDye kbA-dvU"
      >
        l
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
